### PR TITLE
Add Placeholder and Group apis to the frontend

### PR DESCRIPTION
**Pull request by b12f**

### CODE REVIEW

File: frontend/src/app/modules/apiv3/endpoints/groups/apiv3-groups-paths.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      .halResourceService
+      .get<CollectionResource<GroupResource>>(this.path + listParamsString(params));
+  }
+
+  /**
+   * Create a new GroupResource
+   *
+   * @param resource
+   */
+  public post(resource:{ name:string }):Observable<GroupResource> {
```
Comment (b12f) [inline frontend/src/app/modules/apiv3/endpoints/groups/apiv3-groups-paths.ts:64]: Not sure if this is optimal, and we have a similar problem below; we don't send a full HalResource into the API, so we get this dangling typing that's specific to this call, but so far has no real "name". We could make this into an interface, but I guess this does the type checking fine.

File: frontend/src/app/modules/apiv3/endpoints/groups/apiv3-group-paths.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+export class Apiv3GroupPaths extends APIv3GettableResource<GroupResource> {
+  /**
+   * Update a placeholder user resource or payload
+   * @param resource
+   */
+  public patch(resource:GroupResource|Object):Observable<GroupResource> {
+    return this
+      .halResourceService
+      .patch<GroupResource>(this.path, resource);
```
Comment (oliverguenther) [inline frontend/src/app/modules/apiv3/endpoints/groups/apiv3-group-paths.ts]: This probably won't work as you're posting the entire group resource class. You'll need to extract writable properties here (that's what we use the form/schema for in other APIs that support it).

As we don't have forms here, you'll need to build the payload object yourself. You can just add `{ name: resource.name }` as a start, not sure what other properties are writable right now.

File: frontend/src/app/modules/apiv3/endpoints/groups/apiv3-groups-paths.ts (hunk 1 above)
Comment (oliverguenther) [reply frontend/src/app/modules/apiv3/endpoints/groups/apiv3-groups-paths.ts:64]: This is fine for me for now

File: frontend/src/app/modules/apiv3/endpoints/placeholder-users/apiv3-placeholder-user-paths.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+export class Apiv3PlaceholderUserPaths extends APIv3GettableResource<PlaceholderUserResource> {
+  /**
+   * Update a placeholder user resource or payload
+   * @param resource
+   */
+  public patch(resource:PlaceholderUserResource|Object):Observable<PlaceholderUserResource> {
+    return this
+      .halResourceService
+      .patch<PlaceholderUserResource>(this.path, resource);
```
Comment (oliverguenther) [inline frontend/src/app/modules/apiv3/endpoints/placeholder-users/apiv3-placeholder-user-paths.ts]: Same comment as for the groups

File: frontend/src/app/modules/apiv3/endpoints/users/apiv3-users-paths.ts [hunk 4]
```diff
@@ -41,4 +42,18 @@ export class Apiv3UsersPaths extends APIv3ResourceCollection<UserResource, APIv3
 
   // /api/v3/users/me
   public readonly me = this.path + '/me';
+
+  /**
+   * Create a new UserResource
+   *
+   * @param resource
+   */
+  public post(resource:UserResource|{firstName:string, email:string, status:'invited'}):Observable<UserResource> {
```
Comment (oliverguenther) [inline frontend/src/app/modules/apiv3/endpoints/users/apiv3-users-paths.ts]: Same comment as above. We can freely choose whether we want to pass a HAL resource here or the payload object itself. I'd vote for the payload object to make the interface clear

**Review by oliverguenther — APPROVED** [review]

**Pull request merged**